src/object/class_object.c

```diff
@@ -2183,6 +2183,28 @@ classobj_has_class_unique_constraint (SM_CLASS_CONSTRAINT * constraints)
   return false;
 }
 
+/*
+ * classobj_has_class_rk_constraint ()
+ *   return: true if a replication key constraint is contained in the constraint list,
+ *           otherwise false.
+ *   constraints(in): constraint list
+ */
+bool
+classobj_has_class_rk_constraint (SM_CLASS_CONSTRAINT * constraints)
+{
+  SM_CLASS_CONSTRAINT *c;
+
+  for (c = constraints; c != NULL; c = c->next)
+    {
+      if (SM_IS_CONSTRAINT_UNIQUE_FAMILY (c->type) && sm_has_non_null_attribute (c->attributes))
```

Comment:
PK의 경우 not null이 보장되기 때문에 sm_has_non_null_attribute() 호출을 안할 수 있도록 다음 조건을 추가하면 좋을 것 같습니다.
```suggestion
      if ((c->type == SM_CONSTRAINT_PRIMARY_KEY) || (SM_IS_CONSTRAINT_UNIQUE_FAMILY (c->type) && sm_has_non_null_attribute (c->attributes)))
```

Reply:
@hornetmj 반영하였습니다. 수정 전, 후 둘다 더블체크인것같긴한데 추후 새로운 매크로를 추가할만한지 검토해보려합니다.

Reply:
@hornetmj 추가로 조건식이 많아 IS_HA_REPLICATION_KEY_CONSTRAINT라는 매크로를 추가했습니다.